docker/dockerfile-build-ubuntu-16.04

```diff
@@ -0,0 +1,79 @@
+FROM ubuntu:16.04
+MAINTAINER Kent Knox <kent.knox@amd>
+
+# Parameters related to building hcc-lc
+ARG rocm_install_path=/opt/rocm
+ARG rocm_build_path=/usr/local/src/hcc-lc
+ARG build_type=Release
+
+# Download and install an up to date version of cmake, because compiling
+# LLVM has implemented a requirement of cmake v3.4.4 or greater
+ARG cmake_prefix=/opt/cmake
+ARG cmake_ver_major=3.7
+ARG cmake_ver_minor=3.7.2
+
+# Install Packages
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends curl && \
+    curl -sL http://packages.amd.com/rocm/apt/debian/rocm.gpg.key | apt-key add - && \
+    sh -c 'echo deb [arch=amd64] http://packages.amd.com/rocm/apt/debian/ xenial main > /etc/apt/sources.list.d/rocm.list' && \
+    apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+    sudo \
+    file \
+    build-essential \
+    git \
+    software-properties-common \
+    wget \
+    python \
+    rpm \
+    g++-multilib \
+    libncurses5-dev \
+    libelf-dev \
+    libc++abi-dev \
+    libc++-dev \
+    hsa-rocr-dev && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# This installs the new cmake side-by-side with the old cmake using update-alternatives
```

Comment:
It's true; updating cmake for 16.04 is not necessary.  However, i wrote the cmake update logic anyways for 14.04, then I thought it might as well be used in 16.04 just to have the latest.  I don't clobber the cmake installed on the system; i install to /opt and use update-alternatives to default to the newer cmake.  If the shipping cmake is desired, update-alternatives could be used to switch back to the distro cmake.

If you'd rather not suffer the compile times on image build, it could be ripped out safely.